docs/aws/audit/redshiftmonitoring/rules/redshift_reserved_node_expiry_in_7_days_remediation.mdx

```diff
@@ -1,6 +1,259 @@
+   - Set up alerts to notify you when a reserved node is approaching its expiration date.
+
+4. **Plan for Renewal or Purchase:**
+   - Based on the expiration information, plan to either renew the expiring reserved nodes or purchase new ones.
+   - Ensure you have the necessary budget and approvals in place to make the purchase or renewal decision promptly.
+
+By following these steps, you can proactively manage your Redshift reserved nodes and avoid unexpected lease expirations.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent Redshift Reserved Node Lease Expiration in the next 7 days using AWS CLI, you can follow these steps:
+
+1. **Check Current Reserved Nodes:**
+   First, you need to list all your reserved nodes to identify which ones are nearing expiration.
+   ```sh
+   aws redshift describe-reserved-nodes
+   ```
+
+2. **Set Up CloudWatch Alarms:**
+   Create a CloudWatch alarm to monitor the expiration of reserved nodes. This will notify you before the lease expires.
+   ```sh
+   aws cloudwatch put-metric-alarm --alarm-name "RedshiftReservedNodeExpiration" \
+   --metric-name "ReservedNodeExpiration" --namespace "AWS/Redshift" \
+   --statistic "Maximum" --period 86400 --threshold 1 --comparison-operator "GreaterThanOrEqualToThreshold" \
+   --evaluation-periods 7 --alarm-actions "arn:aws:sns:your-region:your-account-id:your-sns-topic"
+   ```
+
+3. **Automate Renewal with Lambda:**
+   Create a Lambda function that automatically renews the reserved nodes. First, create a Lambda function and then use the following command to add a trigger for the CloudWatch alarm.
+   ```sh
+   aws lambda create-function --function-name RenewRedshiftReservedNodes \
+   --runtime python3.8 --role arn:aws:iam::your-account-id:role/your-lambda-role \
+   --handler lambda_function.lambda_handler --zip-file fileb://function.zip
+   ```
+
+4. **Schedule Regular Checks:**
+   Use AWS CLI to create a scheduled event that triggers the Lambda function to check and renew reserved nodes regularly.
+   ```sh
+   aws events put-rule --schedule-expression "rate(1 day)" --name "CheckRedshiftReservedNodes"
+   aws events put-targets --rule "CheckRedshiftReservedNodes" --targets "Id"="1","Arn"="arn:aws:lambda:your-region:your-account-id:function:RenewRedshiftReservedNodes"
+   ```
+
+By following these steps, you can proactively monitor and renew your Redshift reserved nodes, preventing lease expiration issues.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Redshift Reserved Node Lease Expiration in the next 7 days using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK for Python (Boto3):**
+   - Install Boto3 if you haven't already.
+   - Configure your AWS credentials.
+
+   ```bash
+   pip install boto3
+   ```
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon Redshift
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>